Cargo.toml

```diff
@@ -28,6 +28,7 @@ toml = "~0.2.1"
 env_logger = "~0.3.4"
 typed-arena = "~1.2.0"
 clap = "~2.19.0"
+parking_lot = "~0.3.6"
```

Comment:
I would prefer using the std lib primitives since they support poisoning. While parking lot is probably a better implementation, I fear that having unusual (c.f., standard Rust) semantics in edge cases is asking for trouble.